tests/TestCase/Database/TypeTest.php

```diff
@@ -159,6 +162,23 @@ public function testReMapAndBuild()
         $this->assertInstanceOf($barType, $type);
     }
 
+    /**
+     * Tests new types can be registered and built as objects
+     *
+     * @return void
+     */
+    public function testMapAndBuildWithObjects()
+    {
+        $map = Type::map();
+        Type::clear();
+
+        $uuidType = new UuidType('uuid');
+        Type::map('uuid', $uuidType);
+
+        $this->assertSame($uuidType, Type::build('uuid'));
+        Type::map($map);
```

Comment:
unnecessary  (setup() and tearDown() ... )
but I have another concern

Reply:
Thats the only way to reset..

Reply:
my concern is about `$_builtTypes`
you want to reset `$_types` by  `map($map)`  but `$_builtTypes` remains 
and in next usage that use old `$_builtTypes` :)

Reply:
Are you sure?
The code reads

    static::$_types = [];
    static::$_builtTypes = [];

for clear()

Reply:
You are right, I now removed the caching and deprecated the object usage here.
Should be good now.

Reply:
see map and build 
```php
    public static function map($type = null, $className = null)
    {
        if ($type === null) {
            return static::$_types;
        }
        if (is_array($type)) {
            static::$_types = $type;

            return null;
        }

/////////
    public static function build($name)
    {
        if (isset(static::$_builtTypes[$name])) {
            return static::$_builtTypes[$name];
        }
```
if you need to reset `$_builtTypes`
try 
```php
$map = Type::map();
Type::clear();
Type::map($map);
```
in the setup and teardown()